cmake/Modules/FindAWSSDK_EP.cmake

```diff
@@ -102,7 +102,7 @@ if (NOT AWSSDK_FOUND)
       set(CONDITIONAL_PATCH patch -N -p1 < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_awssdk/awsccommon.patch &&
                             patch -N -p1 < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_awssdk/awsconfig_cmake_3.22.patch)
     endif()
-    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR NOT WIN32)
```

Comment:
You're right, `NOT MSVC` would be equivalent, and better, because clang can be used on windows (in theory) and would still have the deprecation error. I'll update it with the other PR.